src/averageAges.js

```diff
@@ -15,11 +15,27 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+  let result = 0;
+
+  if (!century) {
+    const ageOfDeath = men.map(person => person.died - person.born);
+    const sumAge = ageOfDeath.reduce((sum, curent) => sum + curent, 0);
+
+    result = sumAge / men.length;
+  } else {
+    const startCenture = century - 1 + '01';
+    const endCenture = century + '00';
+    const centureResultArr = men.filter(pers =>
+      pers.died > startCenture && pers.died < endCenture
+    );
+    const ageOfDeath = centureResultArr.map(pers => pers.died - pers.born);
+    const sumAge = ageOfDeath.reduce((sum, curent) => sum + curent, 0);
+
+    result = sumAge / centureResultArr.length;
+  }
```

Comment:
let's remove if else with century
add this condition inside filter - find all men or find all men who died in certain venture when century is passed (you can use ternary operator)